src/NHibernate/Logging.cs

```diff
@@ -76,7 +76,24 @@ public interface IInternalLogger
 		void WarnFormat(string format, params object[] args);
 	}
 
-	public interface INHibernateLogger
+	// Since v5.1
+	// Required till look alike methods taking object are dropped.
+	[Obsolete("Implement and use NHibernate.INHibernateLogger")]
+	public interface ITransitionalInternaLogger : IInternalLogger
+	{
+		void Fatal(string message);
+		void Error(string message);
+		void Warn(string message);
+		void Info(string message);
+		void Debug(string message);
+	}
+
+#pragma warning disable 618 // ITransitionalInternaLogger is obsolete, to be removed in a upcoming major version
+	/// <summary>
+	/// NHibernate internal logger interface.
+	/// </summary>
+	public interface INHibernateLogger: ITransitionalInternaLogger
```

Comment:
The new interface is still a `IInternalLogger` for avoiding a breaking change.
For allowing transitioning from obsolete methods to non obsolete ones, which are mostly extensions methods, I was forced for the "just a string" log case to introduce additional not yet obsolete methods, but which will be drop-able without breaking compilation (still a binary break of course) once the other obsoletes are dropped. Thus the intermediate interface `ITransitionalInternalLogger` already obsolete.